include/triton/Dialect/Triton/IR/TritonTypes.td

```diff
@@ -26,7 +26,7 @@ def TT_BoolLike : AnyTypeOf<[I1, TT_BoolTensor]>;
 
 // Integer Type
 def I4 : I<4>;
-def TT_Int : AnyTypeOf<[I1, I4, I8, I16, I32, I64], "integer">;
+def TT_Int : AnyTypeOf<[I1, I4, I8, I16, I32, UI32, I64], "integer">;
```

Comment:
I don't think we want to allow signed/unsigned types in our IR. In general those MLIR types are meant to be used by front end and should be lowered before lowering to LLVM. Since we don't  use those types at all I don't think we want to start introducing them here.
I think we should look at a different solution if we want to handle this case. Could you describe the set of cases you want to handle?

Reply:
Thanks for taking a look! 

> Could you describe the set of cases you want to handle?

The case is mostly to print original negative values as negative (instead of unsigned); the PR description has the minimal reproducer case (and this commit https://github.com/triton-lang/triton/pull/4363/commits/9cd72c21a5b497e6fe62d2e08f8bd43b922e6cab has the python unit tests for reproducing this issue as well)

Happy to hear what you think could be the right solution!

Reply:
I'd prefer we add attributes to the PrintOp to tell how we want each Value to be printed. This would allow us to potentially scale it to different modes and it saves us having to deal with a new type in the IR.